### DISCUSSION TITLE
Link not working inside AutoCompleteItem

### DISCUSSION THREAD

**Discussion by srivatsa17** [heroui-inc/heroui#2086] [Help] [answered]
I am trying to do something like this inside the AutoComplete component but my link is not working. 
Also i am getting warning like this on the console `<Item> with non-plain text contents is unsupported by type to select for accessibility. Please add a textValue prop.`
How can i fix this ? 

```
import { Autocomplete, AutocompleteItem, Link } from "@nextui-org/react";

<AutoComplete
 ...
>
  {(category) => (
    <AutocompleteItem
      key={category.name ?? ""}
      value={category.name}
      className="capitalize"
    >
      <Link
        href={`/${category.slug}`}
        underline="none"
        color="foreground"
      >
        {category.name}
      </Link>
    </AutocompleteItem>
  )}
</AutoComplete>
```

**Comment by isaacdarcilla** [comment]
You can do something like this.

```tsx
import { Autocomplete, AutocompleteItem, Link } from "@nextui-org/react";

<AutoComplete
 ...
>
  {(category) => (
    <AutocompleteItem
      key={category.name ?? ""}
      value={category.name}
      className="capitalize"
      onSelectionChange={() => {
         router.push('link here')
      }}
    >
        {category.name}
    </AutocompleteItem>
  )}
</AutoComplete>
```

**Comment by srivatsa17** [comment, marked as answer]
> You can do something like this.
> 
> ```tsx
> import { Autocomplete, AutocompleteItem, Link } from "@nextui-org/react";
> 
> <AutoComplete
>  ...
> >
>   {(category) => (
>     <AutocompleteItem
>       key={category.name ?? ""}
>       value={category.name}
>       className="capitalize"
>       onSelectionChange={() => {
>          router.push('link here')
>       }}
>     >
>         {category.name}
>     </AutocompleteItem>
>   )}
> </AutoComplete>
> ```

Thanks for the response! I tried this and its working now. I don't know if its the right approach though
```
{(category) => (
    <AutocompleteItem
        ...
        as={Link}
        href=`my link`
    >
        {category.name}
    </AutocompleteItem>
)}
```

**Comment by Arshia-123** [comment]
I'm facing the exact same issue but when placing the Link within a NavbarItem, seems like Links don't work well with items.

I keep getting this warning aswell,

console.js:213 <Item> with non-plain text contents is unsupported by type to select for accessibility. Please add a `textValue` prop.

I fixed it by using OnClick() and useNavigate but its an issue which should be checked out.

**Comment by robpetty** [comment]
Is there a better understanding why this happens and the fix?  Basically the "fix" is pass in a prop that is just text.

However when you want to make a custom view this won't work.  For example I am using <AutoComplete> and in the <AutocompleteItem ... /> I have some TSX and get this same error:

<Item> with non-plain text contents is unsupported by type to select for accessibility. Please add a `textValue` prop.

In my case, it still pops up, see the choices, can select and even the choice has the little selected icon on it.  However, it does prevent the "at-rest" view from working correctly.  It will still show your placeholder value and not that value of what you just selected.

So more broadly, since this use case is not about an action (like clicking a button) but rather just custom display of data, whats the workaround?  Even if you take the example from the docs:

https://nextui.org/docs/components/autocomplete#custom-items

And strip it down, it has same issue.  Even a simple <div>test</div> will replicate it (as content of the <AutocompleteItem />.

Here is a simple repo:
```
<Autocomplete
  label="Make a choice"
  className="mb-2"
  placeholder="Your choices"
>
  <AutocompleteItem key="1" value="1">
    <div className="flex flex-row">test</div>
  </AutocompleteItem>
  <AutocompleteItem key="2" value="2">
    <div className="flex flex-row">test2</div>
  </AutocompleteItem>
</Autocomplete>
```
<img width="1088" alt="image" src="https://github.com/nextui-org/nextui/assets/1473437/fb35fe44-485e-4181-b00e-70fa6530c0cd">

<img width="491" alt="image" src="https://github.com/nextui-org/nextui/assets/1473437/8d143d05-366a-4573-ae64-9d84f4e12d68">

<img width="474" alt="image" src="https://github.com/nextui-org/nextui/assets/1473437/34289262-e301-4724-8e08-98e0299eeb49">

Maybe I should open a bug on this?

**Comment by robpetty** [comment]
Ok, the answer was simple, the road to it was not.  After posting I hit the discord and then the docs one more time, and totally realized I did not have textValue set.  Normally I don't set it all, and that is because it seems you do not need to in most cases because the "normal" case is, you pass text value into the child.  When you pass in TSX/JSX however, it can't infer its textual value and gives the error.  Simply adding the attribute fixes the issue:

<img width="240" alt="image" src="https://github.com/nextui-org/nextui/assets/1473437/05f9e079-735a-4f5f-a03b-103f0e53f662">

**Reply by timsntech** [reply]
@robpetty worked for me, thanks!
